### PR TITLE
[MOD-3008]: Add friendly error message when attempting to use `modal shell` on windows

### PR DESCRIPTION
**Pull request by kramstrom**
`modal shell` command doesn't work on windows currently, this PR adds a check to see which platform is being executed on and raises a friendly message if it's not compatible

### CODE REVIEW

**Review by mwaskom — APPROVED** [review]
Nice — do we need this for `modal container exec` too?

File: test/cli_test.py [hunk 1]
```diff
@@ -439,6 +439,20 @@ def test_shell_cmd(servicer, set_env_client, test_dir, mock_shell_pty):
     assert captured_out == [(1, shell_prompt), (1, expected_output)]
 
 
+@pytest.mark.parametrize(("platform", "expected_exit_code"), [("win32", 1), ("cygwin", 1), ("linux", 0), ("darwin", 0)])
```
Comment (erikbern) [inline test/cli_test.py:442]: we actually run the tests in CI on linux, windows, and osx, so another option is to skip the mocking and parametrization and make sure the behavior is expected on each platform

Comment (mwaskom) [reply test/cli_test.py:442]: I was going to suggest this too but don't understand the difference between `win32` and `cygwin` or whether our CI would exercise both 🪟 🙈

Comment (kramstrom) [reply test/cli_test.py:442]: Yeah I was wondering that too, I'll make a quick test to see what we're running in the cI

Comment (kramstrom) [reply test/cli_test.py]: Tests seem to run on win32: https://github.com/modal-labs/modal-client/actions/runs/9940540142/job/27457485407?pr=2006 but I think it should be fine to keep skip the mocking and run it on all platforms?

**Comment by kramstrom** [comment]
> Nice — do we need this for `modal container exec` too?

Ah didn't know about that one, buy yeah makes sense there too

**Pull request merged**